test/test_infra.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2020 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import io
+import itertools
+import os
+import re
+import unittest
+from unittest.mock import patch
+
+import autopep8
+from flake8.api.legacy import get_style_guide
+
+
+def get_python_filepaths():
+    """Helper to retrieve paths of Python files."""
+    python_paths = ['setup.py']
+    for root in ['ops', 'test']:
+        for dirpath, dirnames, filenames in os.walk(root):
+            for filename in filenames:
+                if filename.endswith(".py"):
+                    python_paths.append(os.path.join(dirpath, filename))
+    return python_paths
+
+
+class InfrastructureTests(unittest.TestCase):
+
+    def test_autopep8(self):
+        # give hints to the developer on how files' style could be improved
+        options = autopep8.parse_args([''])
+        options.aggressive = 1
+        options.diff = True
+        options.max_line_length = 99
+
+        issues = []
+        for filepath in get_python_filepaths():
+            diff = autopep8.fix_file(filepath, options=options)
+            if diff:
+                issues.append(diff)
+
+        if issues:
+            self.fail("Please fix files as suggested:\n" + "\n".join(issues))
+
+    def test_flake8(self):
+        # verify all files are nicely styled
+        python_filepaths = get_python_filepaths()
+        style_guide = get_style_guide()
+        fake_stdout = io.StringIO()
+        with patch('sys.stdout', fake_stdout):
+            report = style_guide.check_files(python_filepaths)
+        self.assertEqual(
+            report.total_errors, 0, "There are flake8 issues!\n" + fake_stdout.getvalue())
```

Comment:
If we feel autopep8 doesn't provide value above flake8, could we get flake8 to tell us what files to go look at and only run autopep8 on those?

Reply:
This is a good idea! we keep the benefits of autopep8 telling us how to fix stuff, but we speed everything up by not running it where we don't care.